src/app/fdctl/run/tiles/fd_dedup.c

```diff
@@ -172,22 +187,13 @@ after_frag( void *             _ctx,
                     payload_sz, txn_t_sz ));
     }
 
-    /* Assert that the payload_sz includes all signatures.
-       We don't need them all for dedup. */
-    ushort sig_off = txn->signature_off;
-    ulong sig_end_off = sig_off + FD_TXN_SIGNATURE_SZ * txn->signature_cnt;
-    if( FD_UNLIKELY( sig_end_off > payload_sz ) ) {
-      FD_LOG_ERR( ("txn is invalid: payload_sz = %lx, sig_off = %x, sig_end_off = %lx", payload_sz, sig_off, sig_end_off ) );
-    }
-
-    /* Write signature for dedup. */
-    ulong txn_sig = FD_VERIFY_DEDUP_TAG_FROM_PAYLOAD_SIG( dcache_entry + sig_off );
-    *opt_sig = txn_sig;
-
     /* Write new size for mcache. */
     *opt_sz = new_sz;
   }
 
+  /* Compute hashmap(signature) for dedup. */
+  *opt_sig = fd_hashmap_hash( &ctx->hashmap, dcache_entry + txn->signature_off, 64UL );
```

Comment:
Should this just be `ulong sighash = fd_hashmap_hash` ? Seems unclear to use `*opt_sig` as a temporary variable here

Reply:
it's not temporary, it's used below... changed the code to make it more clear

Reply:
fixed